optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/inverserelation/InverseRelationShadowVariableDescriptor.java

```diff
@@ -86,19 +86,19 @@ private void linkShadowSources(DescriptorPolicy descriptorPolicy) {
         sourceVariableDescriptor = sourceEntityDescriptor.getVariableDescriptor(sourceVariableName);
         if (sourceVariableDescriptor == null) {
             throw new IllegalArgumentException("The entityClass (" + entityDescriptor.getEntityClass()
-                    + ") has a @" + InverseRelationShadowVariable.class.getSimpleName()
+                    + ") has an @" + InverseRelationShadowVariable.class.getSimpleName()
                     + " annotated property (" + variableMemberAccessor.getName()
                     + ") with sourceVariableName (" + sourceVariableName
                     + ") which is not a valid planning variable on entityClass ("
                     + sourceEntityDescriptor.getEntityClass() + ").\n"
-                    + entityDescriptor.buildInvalidVariableNameExceptionMessage(sourceVariableName));
+                    + sourceEntityDescriptor.buildInvalidVariableNameExceptionMessage(sourceVariableName));
```

Comment:
Wrong entity descriptor was used to build the error message. Same issue as above.